samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -247,4 +220,70 @@ private void verifyNumMessages(String topic, int expectedNumMessages) {
 
     Assert.assertEquals(count, expectedNumMessages);
   }
+
+  private static class Mocks {
+    CountDownLatch latch;
+    KafkaConsumer consumer;
+    KafkaProducer producer;
+    StreamProcessor processor;
+    StreamProcessorLifecycleListener listener;
+
+    private Mocks(String bootstrapServer) {
+      latch = new CountDownLatch(1);
+      initProcessListener();
+      initConsumer(bootstrapServer);
+      initProducer(bootstrapServer);
+    }
+
+    Mocks(Config config, StreamTaskFactory taskFactory, String bootstrapServer) {
+      this(bootstrapServer);
+      processor = new StreamProcessor(config, new HashMap<>(), taskFactory, listener);
+    }
+
+    Mocks(Config config, AsyncStreamTaskFactory taskFactory, String bootstrapServer) {
+      this(bootstrapServer);
+      processor = new StreamProcessor(config, new HashMap<>(), taskFactory, listener);
+    }
+
+    private void initConsumer(String bootstrapServer) {
+      consumer = TestUtils.createNewConsumer(
+          bootstrapServer,
+          "group",
+          "earliest",
+          4096L,
+          "org.apache.kafka.clients.consumer.RangeAssignor",
+          30000,
+          SecurityProtocol.PLAINTEXT,
+          Option$.MODULE$.empty(),
+          Option$.MODULE$.empty(),
+          Option$.MODULE$.empty());
+    }
+
+    private void initProcessListener() {
```

Comment:
nit: perhaps rename to initProcessorListener

Reply:
done